eth/beacon/state_machines/base.py

```diff
@@ -83,25 +83,30 @@ def get_block_class(cls) -> Type[BaseBeaconBlock]:
             return cls.block_class
 
     @classmethod
-    @functools.lru_cache(maxsize=32)
     @to_tuple
-    def get_prev_blocks_to_slot(cls,
-                                last_block_hash: Hash32,
-                                chaindb: BaseBeaconChainDB,
-                                depth: int,
-                                start_slot: int) -> Iterable[BaseBeaconBlock]:
+    def get_prev_blocks(cls,
+                        last_block_hash: Hash32,
+                        chaindb: BaseBeaconChainDB,
+                        searching_depth: int,
+                        target_slot_number: int) -> Iterable[BaseBeaconBlock]:
+        """
+        Returns the previous blocks.
+
+        Since the slot numbers might be not continuous. The searching depth is ``searching_depth``
+        and the target slot depth is ``target_slot_number``.
```

Comment:
Suggested rephrasing.

```
Slot numbers are not guaranteed to be contiguous since it is possible for there
to be no block at a given slot.  The search is bounded by two parameters. 

- `max_search_depth` - The maximum number of slots below the slot of the block denoted by `last_block_hash` that we should search.
- `min_slot_depth` - The slot number for which we should not search any deeper if reached, though we do include the first block encountered below this slot number before terminating the search.
```